clients/fides-js/src/integrations/gtm.ts

```diff
@@ -38,16 +38,26 @@ const pushFidesVariableToGTM = (fidesEvent: {
  * `dataLayer` under `Fides.consent` variable.
  */
 export const gtm = () => {
+  // List every FidesEventType as a record so that new additional events are
+  // guaranteed to be pushed to GTM.
+  const fidesEvents: Record<FidesEventType, true> = {
+    FidesInitializing: true,
+    FidesInitialized: true,
+    FidesUpdating: true,
+    FidesUpdated: true,
+    FidesUIChanged: true,
+    FidesUIShown: true,
+    FidesModalClosed: true,
+  };
+
+  const events = Object.keys(fidesEvents) as FidesEventType[];
```

Comment:
I would consider bumping this to a constants file that can be imported here and in the test (and anywhere else we need it in the future). Would it make more sense as an array instead of an object? Removes the need to coerce it to an array on the next line.
```suggestion
  const fidesEvents: FidesEventType[] = [
    FidesInitializing,
    FidesInitialized,
    FidesUpdating,
    FidesUpdated,
    FidesUIChanged,
    FidesUIShown,
    FidesModalClosed,
  };
```

Reply:
@gilluminate 

I don't think that an `Array` would do what I'm aiming to do here but I'm not a type system expert. The point of this change wasn't to reduce typing. The point of this change is to make sure that the event list was exhaustive. 

With the way this is written if anyone ever adds a new type to `FidesEventType` what ends up happening is Typescript realizes that this object is missing a key since it MUST be implemented. AFAIK there's no way to write "this array is an exhaustive list of keys for `FidesEventType`".

Reply:
I see what you are saying. I think the more correct thing might be to use an `enum` to define the event types instead of a list of strings. That way you can iterate over the enum here, and any additions/changes to the event list is only happening in 1 place. That will require a bit of refactor though, to use the enum in all of those places instead of a string.

Reply:
I would still have to cast in that case, no? 

<img width="502" alt="image" src="https://github.com/user-attachments/assets/82f5ce54-a56d-4555-ad12-7cac693ab221" />

Also the downside of that approach (as far as I can project) would be that it would make ALL events push to GTM. Maybe there's a reason we would want _some_ events not to go to GTM? I'm not sure. I also didn't want to introduce behavior that would have an author not be aware of the fact that they're changing the GTM integration when adding events. The current structure alerts the author to the existence of the connection to GTM without forcing GTM adoption on them (they could use `Pick<>` etc.)

I can see an argument for "yes we should ALWAYS push every Fides event to GTM" but I was trying to be more circumspect than that.

Reply:
Ok I'm getting it now. I'm on board with the original code now, thanks for walking me through it.